src/Session.ts

```diff
@@ -0,0 +1,151 @@
+import msgpack5 from 'msgpack5';
+import * as cookie from './Cookie';
+import * as cache from './Cache';
+import { loggerInstance } from './Logger';
+
+const _appDataKey = process.env.SESSION_APP_KEY;
+const _defaultTtl = 60 * 60;
+
+const Session: { [k: string]: any } = {
+
+  cookie: cookie,
+
+  cache: cache,
+
+  /**
+   * Set up default parameters
+   *
+   * @param req - the request object as supplied the the consumer
+   * @return <void>
+   */
+  _setUp: function (req): void {
+    try {
+      this.sessionData = { appData: null, accountData: null };
+      this.sessionId = this.cookie.getSessionId(req.cookies);
+    } catch (err) {
+      loggerInstance().error(err);
+    }
+  },
+
+  /**
+   * Bootstrap a session
+   *
+   * @param req - the request object as supplied the the consumer
+   * @param res - the response object as supplied the the consumer
+   * @return <void>
+   */
+  start: function (req, res): Promise<boolean> {
+    this._setUp(req);
+    return Promise.all(
+      [
+        this.read('appData'),
+        this.read('accountData')
+      ]
+    ).then(([a, b]) => {
+      this.sessionData.appData = JSON.parse(a);
```

Comment:
Session const seems to be shared between many consecutive requests. Keeping application or session data inside shared constant does not smell good.

Reply:
Yes, that's the idea. "Sharability" is our aim here. We need the same session data accessible to each request because session data must be sharable across multiple requests for an app, and if that session data changes, the updated data must be available to subsequent requests. The same is true for user session data we get from accounts.

Reply:
This design can lead to user B being able to see data of user A. This is obviously bad. User sessions should be separated from each other, that is why each session ID is different.

Let's talk tomorrow.

Reply:
Forgot to tell you.. that User A/User B issue was fixed :)